### DISCUSSION TITLE
X-trap auto focus bug within modals.

### DISCUSSION THREAD

**Discussion by aaronstezycki** [alpinejs/alpine#3328] [7. Bugs (archived)]
Using x-trap on a modal with scrollable content and buttons causes auto focus on button elements when initially opening the modal. This causes scrollable content to start at the bottom of the modal.

ie. ` x-trap.noscroll="isOpen"`

This bug can be viewed by simply using the Alpine JS modal component, and stuffing the modal with content to create vertical scrolling (using the browser inspector). Then close, and open the modal to see auto scroll bug.

[Alpine Modal](https://alpinejs.dev/component/modal)

Obviously this isn't desirable behaviour, and we need a way of disabling this. 

**Observations:**
It should be disabled by default, but enabled on things like dialog boxes whereby an action is required by users and focus is automatically set to buttons to allow for quick submission (which I think is the intended purpose of this).

**Work Around:**
Currently setting `tabindex="1"` to the first element within the modal can be used as a rudimentary fix, however not ideal.

**Comment by aaronstezycki** [comment]
Obviously, I expected this to be created as an actual Github issue, and not a discussion. 🤷🏻‍♂️

**Reply by SimoTod** [reply]
FYI: https://github.com/alpinejs/alpine/discussions/1423
The tab is more to maintain the history now.

**Reply by aaronstezycki** [reply]
Ah okay, I missed this and just expected it to be regarded as an actual issue rather than a discussion. I imagine discussions can be lost in the ether.

... but having now read the link, I totally understand :)
